src/Filter/ChoiceFilter.php

```diff
@@ -54,62 +66,70 @@ public function getRenderSettings()
         ]];
     }
 
-    private function filterWithMultipleValues(ProxyQueryInterface $queryBuilder, string $alias, string $field, array $data = []): void
+    /**
+     * NEXT_MAJOR: Change the typehint to ProxyQueryInterface.
+     */
+    private function filterWithMultipleValues(BaseProxyQueryInterface $query, string $alias, string $field, array $data = []): void
```

Comment:
It's not related to this PR, but `array $data = []` (the default value) I think it doesn't make sense since next line is checking `count($data['value'])`, it would be nice to change this array to a VO, I'll create an issue.

Reply:
Nice catch. This PR is big enough indeed, so I would prefer an issue indeed. :)